### PR TITLE
feat: Add support for NO_PROXY for websocket connections

### PR DESCRIPTION
**Pull request by jbw-vtl**
In https://github.com/PrefectHQ/prefect/pull/16326 the team added support for HTTP_PROXY & HTTPS_PROXY for the websocket connection used by the agent to submit events (such as task runs).
However in some environments these proxies are necessary to route connection for the flow itself running (i.e. to connect to external in an air gapped environment).
Having the events routed through the proxy might mean they cannot reach the server.

This PR adds basic support for the NO_PROXY variable for the websocket connection, allowing to ignore the proxy for the server connection.

### Checklist
- [x] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- ~~[ ] If this pull request removes docs files, it includes redirect settings in `mint.json`.~~
- [x] If this pull request adds functions or classes, it includes helpful docstrings.

This stemmed from the discussion around https://github.com/PrefectHQ/prefect/issues/15153 and closes https://github.com/PrefectHQ/prefect/issues/16537

### CODE REVIEW

**Comment by jbw-vtl** [comment]
Adjusted the used proxy to be `http` as even a HTTPS_PROXY needs to be `http`, the tests without mocking were failing with the below

```python
def parse_proxy_url(url: str) -> Tuple[ProxyType, str, int, Optional[str], Optional[str]]:
        parsed = urlparse(url)
    
        scheme = parsed.scheme
        if scheme == 'socks5':
            proxy_type = ProxyType.SOCKS5
        elif scheme == 'socks4':
            proxy_type = ProxyType.SOCKS4
        elif scheme == 'http':
            proxy_type = ProxyType.HTTP
        else:
>           raise ValueError(f'Invalid scheme component: {scheme}')  # pragma: no cover
E           ValueError: Invalid scheme component: https
```

File: src/prefect/events/clients.py
Comment (jakekaplan) [inline src/prefect/events/clients.py:89]: I'm inclined to use `from urllib.request import proxy_bypass` here to avoid having our own implementation if possible.

My reading of [NO_PROXY](https://curl.se/libcurl/c/CURLOPT_NOPROXY.html) is that it's either an exact match or substring (as opposed to just a suffix), which would conflict with this as written. I'm not very familiar, so just want to compare notes if you saw something that suggested it should work differently.

File: tests/utilities/test_proxy.py
Comment (jakekaplan) [inline tests/utilities/test_proxy.py:37]: good catch, thank you!

File: src/prefect/events/clients.py
Comment (jbw-vtl) [reply src/prefect/events/clients.py]: Was looking for precisely this, also not super familiar here.
This looks indeed perfect, swapped it out and working as expected.
Pushed a change to swap out just now

**Review by jakekaplan — APPROVED** [review]
@jbw-vtl will need a review from a core maintainer to merge but this LGTM. Thanks for contributing the fix!

**Review by zzstoatzz — APPROVED** [review]
thank you @jbw-vtl !

**Pull request merged**
